src/Vehicle.Rental.Api/Controllers/VehicleInventoryController.cs

```diff
@@ -24,13 +30,52 @@ public IActionResult GetVehicles(int inventoryId)
             return Ok(result);
         }
 
-        [HttpGet("{vehicleId}")]
+        [HttpGet("{vehicleId:int}")]
         public IActionResult GetVehicle(int inventoryId, int vehicleId)
         {
             var result = _dbContext.Vehicles
                 .FirstOrDefault(v => v.VehicleInventoryId == inventoryId && v.Id == vehicleId);
 
             return Ok(result);
         }
+
+        [HttpGet("{vin}")]
+        public IActionResult GetVehicleByVin(int inventoryId,string vin)
+        {
+            var result = _dbContext.Vehicles
+                .Where(v => v.VehicleInventoryId == inventoryId && v.Vin == vin).ToList();
+            if(result.Count() == 0)
+            {
+                return NotFound();
+            }
+
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public IActionResult AddVehicle(int inventoryId, VehicleViewModel vehicle)
+        {
+            Vehicle vehicleDTO = _mapper.Map<Vehicle>(vehicle);
```

Comment:
Implement validations and custom exceptions and this is for all endpoints where required